specs/core/0_beacon-chain.md

```diff
@@ -859,16 +865,17 @@ def split(values: List[Any], split_count: int) -> List[Any]:
     ]
 ```
 
-#### `get_new_shuffling`
+#### `get_shuffling`
 
 ```python
-def get_new_shuffling(seed: Hash32,
-                      validators: List[ValidatorRecord],
-                      crosslinking_start_shard: int) -> List[List[ShardCommittee]]:
+def get_shuffling(seed: Hash32,
+                  validators: List[ValidatorRecord],
+                  crosslinking_start_shard: int,
+                  slot: int) -> List[List[ShardCommittee]]:
     """
     Shuffles ``validators`` into shard committees using ``seed`` as entropy.
     """
-    active_validator_indices = get_active_validator_indices(validators)
+    active_validator_indices = get_active_validator_indices(validators, slot)
```

Comment:
`get_shuffling` maybe should normalize slot to be the start slot of the epoch it is contained within. There are some operations that can make a validator's `exit_slot` not at a precise epoch boundary which would make `get_shuffling` consistent throughout an epoch (and a less useful helper outside of the state transition function).

Another option is to ensure that the `exit_slot` rounds up to the next cycle boundary. This is specifically an issue because `penalize_validator` can be called in the middle of an epoch.

Reply:
I made `get_shuffling` normalize to epoch boundaries.